### PR TITLE
MWPW-136872 first draft for filter blocks

### PR DESCRIPTION
**Pull request by npeltier**
- using sidenav dep,
- using separate spectrum dep,
- introducing unit tests in CC

* Add your specific features or fixes
 
image

Resolves: [[MWPW-136872](https://jira.corp.adobe.com/browse/MWPW-136872)]

**Test URLs:**
- After: https://mwpw-136872--cc--npeltier.hlx.page/drafts/npeltier/plans-card

**(NOTE THIS IS A COMMERCE3 PR)**

### CODE REVIEW

File: creativecloud/blocks/sidenav/sidenav.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+await import('../../deps/sidenav.js');
+loadScript('/creativecloud/deps/merch-spectrum.min.js');
+
+const getValueFromLabel = (label) => label
+  .trim()
+  .toLowerCase()
+  .replace(/(and|-)/g, '')
+  .replace(/\s+/g, '-');
+
+const appendSpSidenavItem = (parent, li) => {
```
Comment (yesil) [inline creativecloud/blocks/sidenav/sidenav.js:12]: `appendSidenavItem` ?  `Sp` is not making it clearer :)

File: creativecloud/blocks/sidenav/sidenav.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { createTag, loadScript } from '../../scripts/miloUtils.js';
+
+await import('../../deps/sidenav.js');
+loadScript('/creativecloud/deps/merch-spectrum.min.js');
```
Comment (yesil) [inline creativecloud/blocks/sidenav/sidenav.js:4]: `merch-spectrum.min.js` should be imported before `sidenav.js` to be on the safe side.
also why not`import` but `loadScript`?

File: package.json [hunk 3]
```diff
@@ -36,5 +39,18 @@
     "stylelint": "14.6.0",
     "stylelint-config-prettier": "9.0.3",
     "stylelint-config-standard": "25.0.0"
+  },
+  "jest": {
```
Comment (yesil) [inline package.json:43]: why jest? let's stick to WTR?

File: test/blocks/sidenav/mocks/body.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<body>
```
Comment (yesil) [inline test/blocks/sidenav/mocks/body.html:1]: maybe call it sidenav.html ? easier to find.

File: creativecloud/blocks/sidenav/sidenav.js (hunk 2 above)
Comment (npeltier) [reply creativecloud/blocks/sidenav/sidenav.js:4]: isn't there a risk that it's imported twice? if not why using loadScript at all in milo?

**Review by 3ch023 — APPROVED** [review]

File: creativecloud/blocks/sidenav/sidenav.js (hunk 2 above)
Comment (yesil) [reply creativecloud/blocks/sidenav/sidenav.js:3]: no imports are always done once, it is safer ;)

Comment (yesil) [reply creativecloud/blocks/sidenav/sidenav.js:3]: import loads ecma script modules,  whereas loadScript can load arbitrary javascript.

File: creativecloud/scripts/miloUtils.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { setLibs } from './utils.js';
```
Comment (yesil) [inline creativecloud/scripts/miloUtils.js:1]: I would not create a specific file only for this.
this should go in https://github.com/npeltier/cc/blob/MWPW-136872/creativecloud/scripts/utils.js

File: creativecloud/deps/sidenav.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  const keyValuePairs = hash.replace(/^#/, "").split("&");
+  for (const pair of keyValuePairs) {
+    const [key, value = ""] = pair.split("=");
+    if (key) {
+      result.push([key, decodeURIComponent(value)]);
+    }
+  }
+  return Object.fromEntries(result);
+}
+function pushState(state) {
```
[the remark this answers is not in the document]
Comment (npeltier) [reply creativecloud/deps/sidenav.js:18]: ok, yes, let's do that later :)

**Review by yesil — APPROVED** [review]

**Pull request merged**